vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -594,6 +595,35 @@ public QueryResultIterator answerQuery(PositiveLiteral query, boolean includeBla
 		return new VLogQueryResultIterator(stringQueryResultIterator, this.correctness);
 	}
 
+	@Override
+	public QueryAnswersSize queryAnswerSize(PositiveLiteral query) {
+		return queryAnswerSize(query, true);
+	}
+
+	@Override
+	public QueryAnswersSize queryAnswerSize(PositiveLiteral query, boolean includeNulls) {
+		validateNotClosed();
+		if (this.reasonerState == ReasonerState.KB_NOT_LOADED) {
+			throw new ReasonerStateException(this.reasonerState, "Querying is not alowed before reasoner is loaded!");
+		}
+		Validate.notNull(query, "Query atom must not be null!");
+
+		final boolean filterBlanks = !includeNulls;
+		final karmaresearch.vlog.Atom vLogAtom = ModelToVLogConverter.toVLogAtom(query);
+
+		long result;
+		try {
+			result = this.vLog.querySize(vLogAtom, true, filterBlanks);
+		} catch (NotStartedException e) {
+			throw new RuntimeException("Inconsistent reasoner state.", e);
+		} catch (NonExistingPredicateException e) {
+			LOGGER.warn("Query uses predicate " + query.getPredicate()
+					+ " that does not occur in the knowledge base. Answer must be empty!");
+			result = 0;
+		}
+		return new QueryAnswersSize(this.correctness, result);

```

Comment:
like in `answerQuery` and `exportQueryAnswersToCsv`, a warning must be logged if Correctness is not `SOUND_AND_COMPLETE`.